src/lib/logger/index.ts

```diff
@@ -0,0 +1,168 @@
+        this.writeLog(output, ELoglevel.INFO, EConsoleType.log);
+        return output;
+    }
+
+    public warn(...args: any[]): string {
+        let output = this.configureOutput(args, ELoglevel.WARN, this.config.transports[0]);
+        this.writeLog(output, ELoglevel.WARN, EConsoleType.warn);
+        return output;
+    }
+
+    public error(...args: any[]): string {
+        let output = this.configureOutput(args, ELoglevel.ERROR, this.config.transports[0]);
+        this.writeLog(output, ELoglevel.ERROR, EConsoleType.error);
+        return output;
+    }
+
+    public setLogLvl(loglevel: ELoglevel) {
+        this.config.transports.forEach((transport) => {
+            transport.loglvl = loglevel;
+        });
+    }
+    //#endregion
+
+    //#region private functions
+    private writeLog(input: string, loglevel: number, logType: number): void {
+
+        for (const transport of this.config.transports) {
+            if (loglevel >= transport.loglvl) {
+                this.writeTypeConsole(input, logType);
+            }
+        }
+    }
+
+    private writeTypeConsole(input: string, logType: number): void {
+        switch (logType) {
+            case EConsoleType.warn:
+                console.warn(input);
+                break;
+            case EConsoleType.error:
+                console.error(input);
+                break;
+
+            default:
+                console.log(input);
+                break;
+        }
+    }
+
+    private setConfig(config?: IClientConfig): IMergedConfig {
+
+        let mergedConfig: IMergedConfig = {
+            transports: []
+        };
+
+        try {
+
+            if (typeof (config) === "undefined") {
+                let defaultTransportConsol = new DefaultConsolTransport();
+                mergedConfig.transports.push(defaultTransportConsol);
+                return mergedConfig;
+            }
+
+            if (config.transports.length === 0) {
+                let defaultTransportConsol = new DefaultConsolTransport();
+                defaultTransportConsol.baseComment = config.baseComment ? config.baseComment : defaultTransportConsol.baseComment;
+                defaultTransportConsol.loglvl = typeof (config.loglvl) !== "undefined" ? config.loglvl : defaultTransportConsol.loglvl;
+                mergedConfig.transports.push(defaultTransportConsol);
+                return mergedConfig;
+            }
+
+            for (const transport of config.transports) {
+                let mergedTransport = new BaseTransport();
+
+                let base = transport.baseComment ? transport.baseComment : (config.baseComment ? config.baseComment : null);
+                let loglvl = typeof (transport.loglvl) !== "undefined" ? transport.loglvl :
+                    (typeof (config.loglvl) !== "undefined" ? config.loglvl : null);
+
+                mergedTransport.baseComment = base === null ? mergedTransport.baseComment : base;
```

Comment:
can you use TypeScript 3.7+ ? In this case you can write here

```ts
mergedTransport.baseComment = base ?? mergedTransport.baseComment
```

Reply:
fixed with https://github.com/senseexcel/ser-ext-ondemand/commit/20834c08ae836aeaa95d2c8de8465dacbb87c2df